src/tests/Common/CLRTest.Execute.Bash.targets

```diff
@@ -313,7 +313,7 @@ fi
 
 $(BashIlrtTestLaunchCmds)
 
-if [ ! -z ${RunCrossGen+x} ]%3B then
+if [ ! -z ${RunCrossGen2+x} ]%3B then
   TakeLock
```

Comment:
It seems we used to take a lock for the old `crossgen` only.  Do we actually need a lock when using `crossgen2`?

Reply:
I believe the lock is useful and it was a bug that it wasn't renamed to use RunCrossGen2 when making Crossgen2 the default. I even have the suspicion I originally put it in myself. The problem is that, as you might remember, the Crossgen2 compilation is a state machine - we first copy the MSIL dlls under IL-CG2 and then we run Crossgen2 to generate the R2R libraries back to the root test folder. In case of concurrently running multiple test cases using the same binary with multiple sets of command-line parameters I believe the absence of the lock can cause a race condition where two test cases try to run Crossgen2 on the same binaries in parallel and one or both may fail due to locked files.

Reply:
I am wondering why we are not hitting this race condition then.

Reply:
So am I. I think that actually the race condition is relatively rare, I just double-checked and the total number of RunOnly tests is just 433 i.e. less than 5% of the total Pri1 test set.